### PR TITLE
src/{lib,error}.rs: add custom Error type

### PR DESCRIPTION
**Pull request by Maxime2**
Add custom Error type supporting NoneError, Unsupported, ReportableBug, Io variants.

### CODE REVIEW

File: src/errors.rs [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+}
+
+impl Eq for Error {}
+
+impl PartialEq for Error {
+    fn eq(&self, other: &Error) -> bool {
+        match *self {
+            // Add here variants for new Error enum members when needed.
+            NoneError => {
+                if let NoneError = *other {
```
Comment (SamuelMarks) [inline src/errors.rs]: as per our discussion, make this a `true` rather than a dual branch

File: src/errors.rs [hunk 2]
```diff
@@ -0,0 +1,139 @@
+use std::{
```
Comment (SamuelMarks) [inline src/errors.rs]: as per our discussion, include a link to sled error with commit hash; to cite them

File: src/errors.rs (hunk 1 above)
Comment (Maxime2) [reply src/errors.rs]: Changed.

File: src/errors.rs (hunk 2 above)
Comment (Maxime2) [reply src/errors.rs]: Added.

**Pull request merged**